plutus-ledger-api/test/Spec/CostModelParams.hs

```diff
@@ -27,23 +26,24 @@ tests =
     "costModelParams"
     [ testCase "length" $ do
             166 @=? length (enumerate @V1.ParamName)
-            166 @=? length v1CostModelParamNames
             175 @=? length (enumerate @V2.ParamName)
-            175 @=? length v2CostModelParamNames
             223 @=? length (enumerate @V3.ParamName)
+            -- FIXME: the following three tests are misleading; better to directly test that
```

Comment:
what's misleading about them? they seem okay? Is it because there could be duplicates or something?

Reply:
_misleading_ is not the right word indeed, I just couldn't come up with a better word.

the _purpose_ of these tests are not clear:

- Is their purpose a regression safety check for the number of expected arguments? Then they are redundant from the tests above.
- Is their purpose to check that `showParamName` is injective? Then it *indirectly* checks that through the length. The actual length is not important, what is important is they are injective and you don't see that directly.

Do I make sense? :D